core-web/libs/sdk/experiments/src/lib/hooks/useExperiments.ts

```diff
@@ -0,0 +1,96 @@
+import { useContext, useEffect } from 'react';
+
+import { isInsideEditor } from '@dotcms/client';
+
+import DotExperimentsContext from '../contexts/DotExperimentsContext';
+import { EXPERIMENT_DEFAULT_VARIANT_NAME, EXPERIMENT_QUERY_PARAM_KEY } from '../shared/constants';
+
+/**
+ * Custom hook `useExperiments`.
+ *
+ * This hook is designed to handle changes in the location of the current DotExperiments
+ * instance and set a global click handler that redirects the application when an element with an
+ * assigned variant is clicked.
+ *
+ * It also manages adding or removing the experimentation query parameter from the URL as
+ * appropriate when a click occurs.
+ *
+ * @returns {void}
+ */
+export const useExperiments = (): void => {
+    const experimentContext = useContext(DotExperimentsContext);
+
+    /**
+     * This `useEffect` hook is responsible for tracking location changes when not inside an editor environment, and invoking the
+     * `locationChanged` method from `experimentContext` with current location and custom redirection function.
+     *
+     */
+    useEffect(() => {
+        if (!experimentContext || typeof document === 'undefined') {
+            return;
+        }
+
+        const insideEditor = isInsideEditor();
+
+        if (!insideEditor) {
+            const location = typeof window !== 'undefined' ? window.location : undefined;
+
+            if (experimentContext && location) {
+                experimentContext.locationChanged(location, experimentContext.customRedirectFn);
+            }
+        }
+    }, [experimentContext]);
+
+    /**
+     * This effect sets a click handler on the document.
+     * It captures click events and redirects to a new URL if the clicked anchor has a variant assigned,
+     * additional removing the experiment query param from the URL.
+     */
+    useEffect(() => {
+        if (!experimentContext) {
+            return;
+        }
+
+        const customClickHandler = (event: MouseEvent) => {
+            let target = event.target as HTMLAnchorElement;
```

Comment:
At his point this could NOT be a `Anchor`

Reply:
Fixed, and fixed to a side case found to
```
 <Link
            href={{
              pathname: '/',
              query: Object.fromEntries(searchParams.entries()) // We need to maintain the query params on the navigation, this way next loads the page with the same query params
            }}>
            <span><span>Home</span></span>  // more than 1 level of deep of the anchor.
          </Link>
```